src/test/java/org/zalando/jackson/datatype/money/MonetaryAmountSerializerTest.java

```diff
@@ -269,4 +273,27 @@ public void shouldSerializeWithType() throws JsonProcessingException {
         assertThat(actual, is(expected));
     }
 
+    @Test
+    public void shouldHandleNullValueFromExpectObjectFormatInSchemaVisitor() throws JsonProcessingException {
+        final MonetaryAmountSerializer monetaryAmountSerializer = new MonetaryAmountSerializer(FieldNames.defaults(), new AmountWriter<BigDecimal>() {
+            @Override
+            public Class<BigDecimal> getType() {
+                return BigDecimal.class;
+            }
+
+            @Override
+            public BigDecimal write(final MonetaryAmount amount) {
+                return amount.getNumber().numberValueExact(BigDecimal.class).stripTrailingZeros();
+            }
+        }, new NoopMonetaryAmountFormatFactory());
+
+        final JsonFormatVisitorWrapper jsonFormatVisitorWrapperMock = mock(JsonFormatVisitorWrapper.class);
+
+        try {
+            monetaryAmountSerializer.acceptJsonFormatVisitor(jsonFormatVisitorWrapperMock, SimpleType.constructUnsafe(javax.money.MonetaryAmount.class));
+        } catch (NullPointerException ex) {
```

Comment:
Should an exception mark the test as failed anyway? Couldn't we just remove the try-catch?

Reply:
yes